hydra/_internal/instantiate/_instantiate2.py

```diff
@@ -67,9 +67,7 @@ def _call_target(_target_: Callable, *args, **kwargs) -> Any:  # type: ignore
 
 
 def _convert_target_to_string(t: Any) -> Any:
-    if isinstance(t, type):
-        return f"{t.__module__}.{t.__name__}"
-    elif callable(t):
+    if callable(t):
         return f"{t.__module__}.{t.__qualname__}"
```

Comment:
According to [`PEP 3155`](https://github.com/facebookresearch/hydra/commit/1122135b11dbcf9e02c2d8ffc6c0785512898417) it is appropriate to look up `__qualname__` on any function or class.